dts/bindings/mtd/nxp,s32-qspi-hyperflash.yaml

```diff
@@ -0,0 +1,115 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  QSPI hyperflash connected to the NXP S32 QSPI bus.
+
+compatible: "nxp,s32-qspi-hyperflash"
+
+include: "nxp,s32-qspi-device.yaml"
+
+properties:
+  device-id-word-addr:
+    required: true
+    type: int
+    description: |
+      The word address of the device ID in ASO (Application-Specific Object).
+      This address specifies the exact location within the memory where the device ID is stored.
+
+  rwds-low-dual-error:
+    type: boolean
+    description: |
+      Enable Read-Write Data Strobe (RWDS) dual error detect.
+
+  secure-region-locked:
+    type: boolean
+    description: |
+      The secure region is locked and cannot be accessed or modified.
+      This is particularly useful in scenarios where sensitive data needs protection from
+      unauthorized access, such as in financial applications or secure communication systems.
+      If it is disable, having access to all memory regions is beneficial during development
+      or debugging phases.
+
+  drv-strength:
+    type: string
+    enum:
+      - DRV STRENGTH 000
+      - DRV STRENGTH 001
+      - DRV STRENGTH 002
+      - DRV STRENGTH 003
+      - DRV STRENGTH 004
+      - DRV STRENGTH 005
+      - DRV STRENGTH 006
+      - DRV STRENGTH 007
+    default: DRV STRENGTH 000
+    description: |
+      Specifies the output driver strength of the device.
+      Typical impedance values for 1.8V:
+        - DRV STRENGTH 000: 27 Ohm
+        - DRV STRENGTH 001: 117 Ohm
+        - DRV STRENGTH 002: 68 Ohm
+        - DRV STRENGTH 003: 45 Ohm
+        - DRV STRENGTH 004: 34 Ohm
+        - DRV STRENGTH 005: 27 Ohm
+        - DRV STRENGTH 006: 24 Ohm
+        - DRV STRENGTH 007: 20 Ohm
+      Typical impedance values for 3V:
+        - DRV STRENGTH 000: 20 Ohm
+        - DRV STRENGTH 001: 71 Ohm
+        - DRV STRENGTH 002: 40 Ohm
+        - DRV STRENGTH 003: 27 Ohm
+        - DRV STRENGTH 004: 20 Ohm
+        - DRV STRENGTH 005: 16 Ohm
+        - DRV STRENGTH 006: 14 Ohm
+        - DRV STRENGTH 007: 12 Ohm
```

Comment:
this needs to be specified properly, eg, using `drive-strength-ohm = <12>;`.

Reply:
it is difficult to use directly the ohm value. we need to separate many case.
Because HAL API requires the input argument that is a register value (range from 0 to 7)
Each option corresponds to a typical impedance value as above.
So I'm using a type string property that map define enum of HAL(eg, 'DRV STRENGTH 007' map QSPI_IP_HF_DRV_STRENGTH_007 and corresponds  12 ohm for 3V).
it is more easy than

Reply:
```
enum {
  DRIVE_STRENGTH_20_OHM = QSPI_IP_HF_DRV_STRENGTH_000,
  DRIVE_STRENGTH_27_OHM = QSPI_IP_HF_DRV_STRENGTH_000,
  DRIVE_STRENGTH_117_OHM = QSPI_IP_HF_DRV_STRENGTH_001,
  DRIVE_STRENGTH_71_OHM = QSPI_IP_HF_DRV_STRENGTH_001,
  ...
}

#define DRIVE_STRENGTH(ohm) DRIVE_STRENGTH_ ## ohm ## _OHM
```

Reply:
This is impossible. So I removed this property.
Because this is an optional setting for flash device, not  mandatory.
We can keep the default reset configuration, not impact to device operation.

Reply:
> This is impossible. So I removed this property.

we just need to add two properties, e.g. `drive-strength-ohm` and `impedance`, and combine them to get the desired HAL enum.

> Because this is an optional setting for flash device, not mandatory.

I'm okay to have the minimal configuration on this pr if it can work with current board supported, it can be extended later on.

> We can keep the default reset configuration (the reset value of register when not setting), not impact to device operation.

where is this default set? it should be documented and I'd also like to see it explicit in the code

Reply:
okay, I added two properties `drive-strength-ohm` and `vcc-mv` for and handle.